### PR TITLE
Postman Collection - Responses

### PR DESCRIPTION
**Pull request by karol-maciaszek**
Implementation of Postman responses transformation.

Note: the Postman Collection feature will be divided into several reviewable parts

### CODE REVIEW

**Comment by chris-miaskowski** [comment]
Not under QA, we only test platform-internal PRs, moving to Review.

File: src/postman/operation.ts [hunk 1]
```diff
@@ -33,12 +34,12 @@ export const transformPostmanCollectionOperation: PostmanCollectionHttpOperation
     path,
     summary: item.name,
     request: transformRequest(item.request),
-    responses: [],
+    responses: item.responses.map(transformResponse),
     /*
     servers: ...,
     security: ...,
     */
-  } as any;
```
Comment (XVincentX) [inline src/postman/operation.ts]: Ah, finally :)

File: src/postman/transformers/request.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export function transformQueryParam(queryParam: QueryParam): IHttpQueryParam {
 
 export function transformHeader(header: Header): IHttpHeaderParam {
   return {
-    name: header.key,
+    name: header.key.toLowerCase(),
```
Comment (XVincentX) [inline src/postman/transformers/request.ts]: I am almost sure that the `Header` object in Postman collection is already returning it in lower case, so you do not have to take care of it. Can you verify and if so, remove this part?

File: src/postman/transformers/request.ts [hunk 3]
```diff
@@ -64,8 +64,8 @@ export function transformBody(body: RequestBody, mediaType?: string): IHttpOpera
   return;
 }
 
-function transformRawBody(raw: string, mediaType: string = 'text/plain'): IMediaTypeContent {
-  if (typeIs.is(mediaType, ['application/json', 'application/*+json'])) {
+export function transformRawBody(raw?: string, mediaType: string = 'text/plain'): IMediaTypeContent {
```
Comment (XVincentX) [inline src/postman/transformers/request.ts]: If there's no body provided, the caller should just not call this function at all, why make the most important argument optional?

File: src/postman/transformers/response.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+export function transformResponse(response: Response): IHttpOperationResponse {
+  const headers = response.headers.map(transformHeader);
+  const mediaType = response.headers.get('content-type');
+
+  return {
+    code: String(response.code),
+    description: response.description && transformDescriptionDefinition(response.description),
+    headers: headers.concat(response.cookies.map(transformCookie).filter((c: Cookie) => c)),
+    contents: mediaType && [transformRawBody(response.body, mediaType)],
```
Comment (XVincentX) [inline src/postman/transformers/response.ts]:
```suggestion
    contents: mediaType && response.body && [transformRawBody(response.body, mediaType)],
```

so we can remove the optional body in the `transformRawBody` function

File: src/postman/transformers/request.ts (hunk 2 above)
Comment (karol-maciaszek) [reply src/postman/transformers/request.ts]: I verified once again and Postman Collection conversion does not touch the case of letters here.

File: src/postman/transformers/request.ts (hunk 3 above)
Comment (karol-maciaszek) [reply src/postman/transformers/request.ts]: Done 👍

File: src/postman/operation.ts (hunk 1 above)
Comment (karol-maciaszek) [reply src/postman/operation.ts]: Aye, I could imagine how it annoys you (and me too).

File: src/postman/transformers/response.ts (hunk 4 above)
Comment (karol-maciaszek) [reply src/postman/transformers/response.ts]: 👍

File: src/postman/transformers/request.ts (hunk 2 above)
Comment (XVincentX) [reply src/postman/transformers/request.ts]: Damn postman library

Comment (karol-maciaszek) [reply src/postman/transformers/request.ts]: Damn indeed

**Pull request merged**